src/TopSurface.h

```diff
@@ -0,0 +1,59 @@
+//Copyright (c) 2017 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#ifndef TOPSURFACE_H
+#define TOPSURFACE_H
+
+#include "utils/polygon.h" //For the polygon areas.
+#include "sliceDataStorage.h" //For the input mesh.
+
+namespace cura
+{
+
+class SliceMeshStorage;
+
+class TopSurface
+{
+public:
+    /*!
+     * Create an empty top surface area.
+     */
+    TopSurface();
+
+    /*!
+     * \brief Generate new top surface for a specific layer.
+     *
+     * The surface will be generated by subtracting the layer above from the
```

Comment:
This way of computing the surface generally leads to top surface areas everywhere where theres a slanted surface.
If you want to remove these small areas you'd need to perform a close operation.

If you want to avoid the double offset required for the close operation you can just use e.g. second wall instead of computing an inset from the outline.

Currently you have an Ironing Inset setting with a default value of .35. You should change it such that the top surface area is computed by subtracting the next layer from the top_bottom_area and performing an offset equal to the Ironing Offset which would have a default value of `wall_line_width_0 + wall_line_width_x * (wall_line_count - 1) - .35`